invenio_rdm_records/assets/semantic-ui/js/invenio_rdm_records/src/deposit/api/DepositApiClient.js

```diff
@@ -322,10 +348,27 @@ export class DepositFileApiClient {
  * Default File API Client for deposits.
  */
 export class RDMDepositFileApiClient extends DepositFileApiClient {
-  initializeFileUpload(initializeUploadUrl, filename) {
+  constructor(additionalApiConfig, defaultTransferType, transferTypes) {
+    super(additionalApiConfig);
+    this.defaultTransferType = defaultTransferType || "L";
+    this.transferTypes = transferTypes || ["L"];
+  }
+
+  initializeFileUpload(initializeUploadUrl, filename, transferOptions = {}) {
+    const { fileSize = 0, type = this.defaultTransferType, ...opts } = transferOptions;
+
+    if (!this.transferTypes.includes(type)) {
+      throw new UnsupportedTransferTypeError(
+        `Unsupported upload TransferType "${type}". Server supports: ${this.transferTypes}`,
```

Comment:
This error seems like an instance misconfiguration. If `transferTypes` is used only for the error message I think just 
```suggestion
        `Unsupported upload TransferType "${type}".
```
should be sufficient.

Reply:
It basically indicates that uploader UI used a transfer type that server isn't configured for. For RDM's Uppy, We can make sure that Uppy instance uses only the server supported transfer types. But devs theoretically will be able to customize the instance via overridable mechanism, e.g. by injecting own plugins. This error shouldn't pop up on the end-users at all